File: Opaline.Core/Auth/OAuthTokens.cs
namespace Opaline.Core.Auth;

public sealed class OAuthTokens
{
    public string AccessToken { get; set; }
    public string RefreshToken { get; set; }
    public string ClientId { get; set; }
    public string ClientSecret { get; set; }
    public DateTimeOffset ExpiresAt { get; set; }

    public bool IsExpired => DateTimeOffset.UtcNow >= ExpiresAt.AddMinutes(-2);
}

File: ios/Opaline/Core/Auth/OAuthTokens.swift
import Foundation

struct OAuthTokens: Codable {
    var accessToken: String
    var refreshToken: String
    var expiryDate: Date
    var clientId: String
    var clientSecret: String
    var isExpired: Bool { expiryDate.timeIntervalSinceNow < 60 }
}

extension Notification.Name {
    static let authorizationRequired = Notification.Name(
        "authorizationRequired"
    )
    static let userDidSignOut = Notification.Name(
        "userDidSignOut"
    )
    static let tokenDidRefresh = Notification.Name(
        "tokenDidRefresh"
    )
    static let showShortsSettingDidChange = Notification.Name(
        "showShortsSettingDidChange"
    )
}
